src/main/java/codeu/model/store/basic/UserStore.java

```diff
@@ -121,4 +128,101 @@ public boolean isUserRegistered(String username) {
   public void setUsers(List<User> users) {
     this.users = users;
   }
+
+  /**
+    * Returns the number of users currently recorded
+    */
+  public int getUserCount() {
+    return users.size();
+  }
+
+  /**
+    * Returns the users array
+    */
+  public List<User> getAllUsers(){
+    return users;
+  }
+
+  /**
+    * Returns the user with the most messages sent
+    * Pre : the the users list should already have the messageCount calculated
+    */
+  public String getTopUser(){
+    sortUserList();
```

Comment:
This is inefficient if you're only getting top 1.  You'd incur the cost of sorting every time this function is called.

There'are a couple of options I'd recommend:

1) Use a sorted collection data structure.  The cost here is that inserts become more expensive.
2)  If getTopUser is called less frequently than users are added, consider just doing a linear scale of users to find the max instead.

Reply:
This would be a different consideration if you needed top N users instead.

Reply:
I see, yeah I think just doing a linear search would be more efficient